src/components/SectionTable/GradesPopup.js

```diff
@@ -0,0 +1,98 @@
+        backgroundColor: '#385EB1',
+        color: '#fff',
+    },
+    gpaTitle: {
+        marginTop: '.5rem',
+        textAlign: 'center',
+        fontWeight: 500,
+        fontSize: '1.2rem',
+        marginRight: '4rem',
+        marginLeft: '4rem',
+    },
+    skeleton: {
+        padding: '4px',
+    },
+};
+
+const GradesPopup = ({ deptCode, courseNumber, classes, isMobileScreen }) => {
+    const [loading, setLoading] = useState(true);
+    const [graphTitle, setGraphTitle] = useState(null);
+    const [gradeData, setGradeData] = useState(null);
+
+    const loadGrades = async () => {
+        if (loading === false) {
+            return;
+        }
+
+        try {
+            let courseGrades = await queryGrades(deptCode, courseNumber);
+
+            let data = [];
+            for (const [key, value] of Object.entries(courseGrades)) {
+                // format data for display in chart
+                // key formatting: sum_grade_a_count -> A
+                if (key !== 'average_gpa') {
+                    data.push({ name: key.split('_')[2]?.toUpperCase(), all: value });
+                }
+            }
+
+            setGraphTitle(`Grade Distribution | Average GPA: ${courseGrades.average_gpa.toFixed(2)}`);
+            setGradeData(data);
+            setLoading(false);
+        } catch (e) {
+            console.log(e);
+            setLoading(false);
+            setGraphTitle('Grades are not available for this class.');
+        }
+    };
+
+    const width = isMobileScreen ? 300 : 500;
+    const height = isMobileScreen ? 200 : 300;
+
+    loadGrades();
+
+    if (loading) {
+        return (
+            <div className={classes.skeleton}>
+                <p>
+                    <Skeleton variant="text" animation="wave" height={height} width={width} />
+                </p>
+            </div>
+        );
+    } else {
+        const encodedDept = encodeURIComponent(deptCode);
+
+        return (
+            <div style={{ marginTop: '5px' }}>
+                <div className={classes.gpaTitle}>{graphTitle}</div>
+                {gradeData && (
+                    <ResponsiveContainer width={width} height={height}>
+                        <BarChart data={gradeData}>
+                            <CartesianGrid strokeDasharray="3 3" />
+                            <XAxis dataKey="name" tick={{ fontSize: 12 }} />
+                            <YAxis tick={{ fontSize: 12 }} width={40} />
+                            <Bar dataKey="all" fill="#5182ed" />
+                        </BarChart>
+                    </ResponsiveContainer>
+                )}
+                <div style={{ marginBottom: '5px', textAlign: 'center' }}>
+                    <CourseInfoButton
+                        text="Zotistics"
+                        redirectLink={`https://zotistics.com/?&selectQuarter=&selectYear=&selectDep=${encodedDept}&classNum=${courseNumber}&code=&submit=Submit`}
+                    />
+                </div>
```

Comment:
This should be an `<a>` tag, not a courseInfoButton

Reply:
Completed. Styling can be done as a fellowship task.
<img width="493" alt="image" src="https://user-images.githubusercontent.com/29494270/173160082-bb669429-a4af-4d8c-81a9-657e7c01845d.png">